source/DistanceAndDirection/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs

```diff
@@ -1071,6 +1071,51 @@ internal void AddTextToMap(IGeometry geom, string text)
             RaisePropertyChanged(() => HasMapGraphics);
         }
 
+        internal void AddTextToMap(IGeometry geom, string text, double angle, AzimuthTypes azimuthType)
+        {
+            var mxDoc = ArcMap.Application.Document as IMxDocument;
+            var av = mxDoc.FocusMap as IActiveView;
+            var gc = av as IGraphicsContainer;
+            double bearing = 0.0;
+            if(azimuthType == AzimuthTypes.Mils)
+            {
+                bearing = angle * 0.05625;
+            }
+            else
+            {
+                bearing = angle;
+            }
+            double rotate = 360 - (bearing + 270.0) % 360;
```

Comment:
minor: could be just (450-bearing) % 360 - also maybe a comment about converting from bearing to XY angle